format/materialized-view-spec.md

```diff
@@ -0,0 +1,55 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Iceberg Materialized View Spec
+
+## Background and Motivation
+Iceberg views are a powerful tool to abstract complex queries and share them among different engines.
+However, such views are not materialized by default, which means that they are re-computed every time they are queried.
+This can be inefficient for complex queries that are computed frequently.
+Iceberg Materialized views are a way to store the results of an Iceberg view to reuse the computation in subsequent queries.
+
+## Goals 
+The goal of this spec is to define the metadata associated with materialized views in Iceberg.
+Such metadata allows creating and querying Iceberg materialized views across different engines.
+
+## Specification
+A materialized view is an Iceberg view with a respective Iceberg table that stores the results of the view query.
+An Iceberg view is considered a materialized view if it has the `iceberg.materialized.view` property set to `true`.
+A materialized view must also reference the storage table identifier in its `iceberg.materialized.view.storage.table` property.
+
+The specification for the materialized view properties on the view is as follows:
```

Comment:
regarding property name iceberg.materialized.view, it is redundant to have iceberg there.

metadata field name uses hyphen `-`. so maybe like `view-materialized` and `view-storage-table`.

Reply:
I have removed the `iceberg` prefix. Not quite sure if there is a rule about about hyphen vs dot. For example, there is a table property: `schema.name-mapping.default` in the table spec.

Reply:
+1 for Steven's suggestion, which is what I also wrote in the comment below. Take `schema.name-mapping.default` as an example, there might be other schema related properties in the future, thus the `schema.` prefix. In general `.` is used for grouping purpose.

Reply:
Ack. I have removed `iceberg.` already. I am thinking of introducing `mv.` now for the grouping purpose.